### DISCUSSION TITLE
Add transaction type #buyDaoCoin

### DISCUSSION THREAD

**Discussion by darian-parrish** [deso-protocol/dips#225] [Large CIPs]
**Purpose:** Enable an automated initial DAO offering and/or continual fundraising.

**How it works:** DAO profile owner mints DAO coins as usual and can set a price for which any user can buy them(price = 0 is valid).  As many DAO tokens that are held by the DAO profile can be purchased and the DeSo used for purchasing goes directly to the DAO wallet(here there is no concept of total value locked, only total value deposited).

**Implementation:** Need a new transaction type. Then can clone the 'buy-or-sell-creator-coin' endpoint and the main new parameter needed is the price at which to sell the DAO coin. The rest can be copied but with no sell parameters.

**Discussion:** This is needed to avoid the complexity and friction of manually collecting $DESO and trying to send the correct amount of coins. We have seen these issues already as creators have tried doing this with creator coins to avoid the bonding curve or get everyone in at the same price but end up making many mistakes.

Furthermore this should happen in a trustless fashion. Can leave the DAO coins marketplace/exchange to the app layer but need the basic protocol mechanism to fund the DAO or garner members.

Without this DAOs are DOA.

**Comment by matt-mcmarsh** [comment]
Sounds good to get some use from DAO soon
